projects/layout/src/lib/components/configurable-layout/configurable-layout.component.ts

```diff
@@ -0,0 +1,214 @@
+	 * Providing both a `control` and the `keys` input will result in the control overriding
+	 * the input.
+	 */
+	@Input() public set keys(keys: string[][]) {
+		// Wouter: If no keys are provided, we prevent the patching of the control.
+		if (!Boolean(keys)) {
+			return;
+		}
+		// Wouter: Patch the provided keys onto the control.
+		this.form.patchValue(keys, { emitEvent: false });
+	}
+
+	/**
+	 * Determines how much space each item takes in the row. By default, this is 'fill'
+	 *
+	 * fill - Will make the items take up the available space
+	 * fit-content - Will make the items take up the space of the provided template
+	 * equal - Will make the items take up equal amount of space throughout the grid
+	 */
+	@Input() public itemSize: NgxConfigurableLayoutItemSizeOption = 'fill';
+
+	/**
+	 * Whether drag and drop is enabled for the grid
+	 */
+	@Input() public allowDragAndDrop: boolean = false;
+
+	// Lifecycle methods
+	// ==============================
+	public ngOnInit(): void {
+		this.form.valueChanges
+			.pipe(
+				tap(() => {
+					this.onChanged(this.form.value);
+					this.onTouched();
+				}),
+				takeUntil(this.destroyed$)
+			)
+			.subscribe();
+	}
+
+	public ngAfterContentChecked(): void {
+		// Iben: Run with content check so that we can dynamically add templates/columns
+		this.handleItemTemplates();
+
+		if (!this.form.value.length) {
+			console.warn(
+				'NgxLayout: Neither a formControl nor a keys input was provided. Therefore no items can be displayed.'
+			);
+		}
+	}
+
+	public ngOnDestroy(): void {
+		this.destroyedSubject.next();
+		this.destroyedSubject.complete();
+	}
+
+	// Component methods
+	// ==============================
+	// TODO: use the ngx-forms formAccessor instead of copying its internal way of working
+	onChanged: Function = () => {};
+	onTouched: Function = () => {};
+
+	writeValue(value: string[][]): void {
+		this.form.setValue(value || [], { emitEvent: false });
+	}
+	registerOnChange(fn: any): void {
+		this.onChanged = fn;
+	}
+	registerOnTouched(fn: any): void {
+		this.onTouched = fn;
+	}
+	setDisabledState?(isDisabled: boolean): void {
+		isDisabled
+			? this.form.disable({ emitEvent: false })
+			: this.form.enable({ emitEvent: false });
+	}
+
+	/**
+	 * Drag and drop an element in the grid
+	 *
+	 * @param {CdkDragDrop<string[]>} event - The grid element
+	 * @memberof NgxConfigurableLayoutComponent
+	 */
+	public drop(event: CdkDragDrop<string[]>) {
+		// Iben: Fetch the arrays we need to update
+		const startArrayIndex = parseInt(event.previousContainer.id.split('ngx-layout-row-')[1]);
+		const endArrayIndex = parseInt(event.container.id.split('ngx-layout-row-')[1]);
+		const formData = [...this.form.value];
+		const startArray = formData[startArrayIndex];
+		const endArray = formData[endArrayIndex];
+
+		// Iben: If the drag and drop is within the same row, we move
+		if (event.previousContainer === event.container) {
```

Comment:
In de docs zie ik dat dit een `CdkDropList<T>` is, kan je die zo vergelijken met elkaar? Kwestie dat je hier niet op reference ah werken bent.

Reply:
Yes, dit is rechtstreeks vanuit de documentatie van de CDK!

Reply:
Ah ok, vreemd wel, maar als het zo gedocumenteerd is is het goed zeker 🤷

Reply:
<img width="1433" alt="afbeelding" src="https://github.com/studiohyperdrive/ngx-tools/assets/10544531/420234dd-4161-4158-921d-0d30dfbcb082">
https://material.angular.io/cdk/drag-drop/examples

Het komt inderdaad uit hun eigen voorbeeld :D